src/logging.c

```diff
@@ -683,14 +661,14 @@ rcutils_ret_t rcutils_logging_format_message(
 # define IS_STREAM_A_TTY(stream) (isatty(fileno(stream)) != 0)
 #endif
 
-#define IS_OUTPUT_COLORIZED(is_colorized, stream) \
```

Comment:
Here and elsewhere, instead of removing the stream, let's leave it configurable and add a new environment variable to let users choose between stderr and stdout.  I don't care hugely what its called, but something descriptive would be nice.  In my opinion, if the environment variable is not specified, the default should be stderr.

Reply:
@clalancette Why should it be reconfigurable here?

* If the user wants all program output in a single stream, that's just `2>&1`.
* If it's about performance, then we should instead change the `stderr` buffering with `setvbuf`.
* If we don't expect stream redirection to be available (or we want to make sure the user can differentiate from other types of program output), we should additionally allow the user to specify an arbitrary file, not just a stream.

Reply:
> @clalancette Why should it be reconfigurable here?

So the 2>&1 isn't that straightforward to achieve when you are launching through a launch file.  While it would be kind of weird to use stdout instead of stderr in one process of a launch file in not another, it isn't invalid.

The other reason to have it is that all of the log libraries I surveyed have the ability to choose which stream to use.  Given that they all have it, presumably it is functionality that is useful in a logging library, so I think we should probably follow suit here.

Reply:
(also, because there is a small performance impact from using stderr, we should give users the ability to opt out of it and get buffered performance if they want)